libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3274,6 +3282,44 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 	});
 }
 
+string YulUtilFunctions::bytesToBytesNNConversionFunction(ArrayType const& _from, FixedBytesType const& _to)
+{
+	solAssert(_from.isByteArray(), "");
+	solAssert(_from.isDynamicallySized(), "");
+	string functionName = "convert_bytes_to_bytesNN_from_" + _from.identifier() + "_to_" + _to.identifier();
+	return m_functionCollector.createFunction(functionName, [&]() {
+		Whiskers templ(R"(
+			function <functionName>(array<?fromCalldata>, len</fromCalldata>) -> value {
+				let length := <arrayLen>(array<?fromCalldata>, len</fromCalldata>)
+				if gt(length, <fixedBytesLen>) { <panic>() }
+				let dataArea := array
+				<?fromMemory>
+					dataArea := <dataLocation>(array)
+				</fromMemory>
+				<?fromStorage>
+					if eq(length, 32) { dataArea := <dataLocation>(array) }
+				</fromStorage>
+				value := <extractValue>(dataArea)
+			}
+		)");
+		templ("functionName", functionName);
+		templ("fromCalldata", _from.dataStoredIn(DataLocation::CallData));
+		templ("arrayLen", arrayLengthFunction(_from));
+		templ("fixedBytesLen", to_string(_to.numBytes()));
+		templ("panic", panicFunction(PanicCode::ResourceError));
```

Comment:
I don't think `ResourceError` is the right category. Maybe we should introduce a new one next to the enum conversion error?

Reply:
I added `FixedBytesConversionError`